codechecker_common/checker_labels.py

```diff
@@ -0,0 +1,262 @@
+        format.
+        """
+        def is_string_list(x):
+            return isinstance(x, list) and \
+                all(map(lambda s: isinstance(s, str), x))
+
+        def is_unique(labels: Iterable[str], label: str):
+            """
+            Check if the given label occurs only once in the label list.
+            """
+            found = False
+            for k, _ in map(self.__get_label_key_value, labels):
+                if k == label:
+                    if found:
+                        return False
+                    else:
+                        found = True
+            return True
+
+        if not isinstance(data, dict):
+            raise ValueError(f'Top level element should be a JSON object.')
+
+        for _, checkers in data.items():
+            for checker, labels in checkers.items():
+                if not is_string_list(labels):
+                    raise ValueError(
+                        f'"{checker}" should be assigned a string list.')
+
+                if any(map(lambda s: ':' not in s, labels)):
+                    raise ValueError(
+                        f'All labels at "{checker}" should be in the '
+                        'following format: <label>:<property>.')
+
+                for unique_label in CheckerLabels.UNIQUE_LABELS:
+                    if not is_unique(labels, unique_label):
+                        raise ValueError(
+                            f'Label "severity" should be unique for checker '
+                            '{checker}.')
+
+    def __get_analyzer_data(self, analyzers: Iterable[str]):
+        for analyzer, checkers in self.__data.items():
+            if analyzers is None or analyzer in analyzers:
+                yield analyzer, checkers
+
+    def checkers_by_labels(
+            self,
+            filter_labels: Iterable[str],
+            analyzers: Iterable[str] = None) -> List[str]:
+        """
+        Returns a list of checkers that have at least one of the specified
+        labels.
+
+        filter_labels -- A string list which contains labels with specified
+                         values. E.g. ['profile:default', 'severity:high'].
+        analyzers -- A list of analyzers among which checkers are searched.
+                     By default all analyzers are searched.
+        """
+        collection = []
+
+        filter_labels = set(map(self.__get_label_key_value, filter_labels))
+
+        for _, checkers in self.__get_analyzer_data(analyzers):
+            for checker, labels in checkers.items():
+                labels = set(map(self.__get_label_key_value, labels))
+
+                if labels.intersection(filter_labels):
+                    collection.append(checker)
+
+        return collection
+
+    def label_of_checker(
+            self,
+            checker: str,
+            label: str,
+            analyzers: Iterable[str] = None) -> Union[str, Iterable[str]]:
```

Comment:
I think this one is more readable:
```suggestion
    def label_of_checker(
        self,
        checker: str,
        label: str,
        analyzers: Iterable[str] = None
    ) -> Union[str, Iterable[str]]:
```
We use this form in other places, so we should use it everywhere. Please change other places too in this file.